lib/covalence/core/state_stores/s3.rb

```diff
@@ -34,12 +34,16 @@ def get_doc(bucket, document)
         @cache[bucket][document] ||= begin
           @s3.get_object(bucket: bucket, key: document).body.read
         rescue Aws::S3::Errors::ServiceError => err
-          fail "Unable to retrieve document '#{document}' from bucket '#{bucket}': " + err.message
+          raise err, err.message + " Unable to retrieve document '#{document}' from bucket '#{bucket}'"
```

Comment:
I would actually override the `err.message` returned by the S3 client entirely instead of appending to it. There are multiple meanings of `key` here, which could lead to confusion.